src/main/frontend/public/js/jsonFetch.js

```diff
@@ -0,0 +1,92 @@
+const jsonGet = async (url) => {
```

Comment:
I love this extraction of common methods into a generic function! Definitely going with the concepts of DRY.  I wonder if it'd be possible to take it one step further?  It seems the only different between the 4 functions is the "method" of the fetch, and the `callback()` when it's not a GET, right?